src/service/url-replacements-impl.js

```diff
@@ -766,6 +786,10 @@ export class GlobalVariableSource extends VariableSource {
    * @private
    */
   getGeo_(getter, expr) {
+    if (this.cachedGeo_ !== null) {
+      return getter(this.cachedGeo_);
+    }
+
     const element = this.ampdoc.getHeadNode();
     return Services.geoForDocOrNull(element).then((geo) => {
       userAssert(geo, 'To use variable %s, amp-geo should be configured', expr);
```

Comment:
nit: could also cache the service here via`this.cachedGeo_ = geo`

Reply:
Added in as well. Thanks for the nit :)